### PR TITLE
Updated import functionality to add contact fields

### PR DESCRIPTION
**Pull request by mdshamoon**
## Summary
 Target issue is #2268 
 
Updated import functionality to add contact fields

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [x] Ran `mix setup` in the repository root and test.
- [x] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

### CODE REVIEW

File: lib/glific.ex [hunk 1]
```diff
@@ -83,6 +83,18 @@ defmodule Glific do
...
+  """
+  @spec string_snake_case(String.t() | nil) :: String.t() | nil
+  def string_snake_case(str) when is_nil(str) or str == "", do: str
+
+  def string_snake_case(str),
+    do:
+      str
+      |> String.replace(~r/\s+/, "_")
+      |> String.downcase()
+
```
Comment (pankaj-ag) [inline lib/glific.ex:97]: Not right now but I think now it's a good time to use a library.  
https://github.com/nurugger07/inflex

File: lib/glific/contacts/import.ex [hunk 2]
```diff
@@ -115,6 +134,32 @@ defmodule Glific.Contacts.Import do
...
+            name: field,
+            shortcode: field,
+            organization_id: first_contact.organization_id
+          })
+
+        contact_field ->
+          {:ok, contact_field}
+      end
+    end)
+
```
Comment (pankaj-ag) [inline lib/glific/contacts/import.ex]: @mdshamoon let's create a function in contacts table with name update_contact_fileds(contact, fields ) 

There are some inconsistencies in the flows also for creating contact fields we will migrate that to this centre function slowly.

File: lib/glific/contacts/import.ex [hunk 3]
```diff
@@ -115,6 +134,32 @@ defmodule Glific.Contacts.Import do
...
+    first_contact = Enum.at(contacts, 0)
+
+    Enum.each(first_contact.contact_fields, fn {field, _value} ->
+      field = Glific.string_snake_case(field)
+
+      case Repo.get_by(Glific.Contacts.ContactsField, %{shortcode: field},
+             organization_id: first_contact.organization_id
+           ) do
+        nil ->
+          ContactField.create_contact_field(%{
```
Comment (pankaj-ag) [inline lib/glific/contacts/import.ex]: let's create a function called may_be_create_contact_fields function and then we can use upsert there.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific.ex (hunk 1 above)
Comment (mdshamoon) [reply lib/glific.ex:97]: Should I update it now only?

File: lib/glific/contacts/import.ex (hunk 3 above)
Comment (mdshamoon) [reply lib/glific/contacts/import.ex]: Done

File: lib/glific/contacts/import.ex (hunk 2 above)
Comment (mdshamoon) [reply lib/glific/contacts/import.ex]: Didn't understand the contacts table?  

Do you mean I need to replace the `do_add_contact_fields` function with `update_contact_fields` function which should be created in contacts.ex file?

File: lib/glific.ex (hunk 1 above)
Comment (pankaj-ag) [reply lib/glific.ex:97]: No No these were just thoughts.

**Review by pankaj-ag — APPROVED** [review]

**Comment by mdshamoon** [comment]
Closing in favor of #2281

**Pull request closed without merging**